src/core/DrumSynth.cpp

```diff
@@ -236,19 +236,11 @@ int DrumSynth::GetPrivateProfileString(
 				}
 				else
 				{
-					k = static_cast<char*>(b + strlen(b) - 1);
-					while ((k >= b) && (*k == ' ' || *k == '\t'))
-					{
-						--k;
-					}
-					*(k + 1) = '\0';
-
-					len = strlen(b);
-					if (len > size - 1)
-					{
-						len = size - 1;
-					}
-					strncpy(buffer, b, len + 1);
+					k = &b[strlen(b) - 1];
+					while ((k >= b) && (*k == ' ' || *k == '\t')) { --k; }
+					k[1] = '\0';
+					len = std::min(static_cast<int>(1 + k - b), size - 1);
```

Comment:
```suggestion
					len = std::min(bLen, size - 1);
```
I think `1 + k - b` is the length of `b` to `k + 1`, which is not the same as the length of `b` to the end of `b`.

It's simpler to just cache the result of `std::strlen(b)` from before and use that.

Reply:
> I think 1 + k - b is the length of b to k + 1, which is not the same as the length of b to the end of b.

Yes. This is trimming off trailing whitespace by placing a new null terminator, so `1 + k - b` is the length of `b` minus the whitespace. Using `bLen` here would be incorrect since the length of the string has changed.